call-home/src/bin/callhome/collector/report_models.rs

```diff
@@ -184,3 +199,164 @@ fn get_pools_size_vector(pools: Vec<openapi::models::Pool>) -> Vec<u64> {
     }
     pools_size_vector
 }
+
+/// EventData consists of pool and volume events.
+#[derive(Debug, Default, Clone)]
+pub(crate) struct EventData {
+    pub(crate) volume_created: VolumeCreated,
+    pub(crate) volume_deleted: VolumeDeleted,
+    pub(crate) pool_created: PoolCreated,
+    pub(crate) pool_deleted: PoolDeleted,
+}
+
+/// Record of events populated from prometheus.
+#[derive(Debug, Default, Clone)]
+/// Wrapper for Stats fetched from prometheus data.
+pub struct EventsRecord {
+    pub record: Vec<Sample>,
```

Comment:
pub(crate) ? or maybe private, this doesn't need to be exposed outside this module?

Reply:
made record as private, cant make `EventsRecord` as pub(crate) as  it is being used here `errors::Result<EventsRecord>` .